### PR TITLE
Handle unicode object ids.

### PR DESCRIPTION
**Pull request by sallner**
This solves #166.

### CODE REVIEW

**Comment by sallner** [comment]
@tseaver There was this test.

https://github.com/zopefoundation/Zope/blob/54c6ea651ca4070ea69b8b7471d8ab32bc034aec/src/OFS/tests/testObjectManager.py#L553-L557

Do you remember why this is bad in the object id? This should be quoted by urllib anyway, shouldn't it?

**Comment by sallner** [comment]
Actually, when using the angle brackets in an object id, it currently fails to do a rename with this patch. I have to look into a that again and produce a failing test first. But it would be still nice to now the reason behind the explicit declaration as illegal. In that case, I would not need to find the reason for the failure.

**Comment by sallner** [comment]
@icemac I added the requested changes and dropped the attempt to allow `<`, `>` and `&`, as they caused an error during a `hasattr` call. The `id` was present as `TaintedString` in this case.

File: src/OFS/tests/testObjectManager.py [hunk 1]
```diff
@@ -528,64 +528,60 @@ def test_empty_string(self):
...
-        # Unicode can only be handled under Python 3, Python 2 needs
-        # bytestrings.
-        if PY2:
-            e = self.assertBadRequest(u'abc☃')
-            self.assertEqual(
-                str(e), "('Empty or invalid id specified', u'abc\\u2603')")
-        else:
-            # Does not raise
-            self._callFUT(self._makeContainer(), u'abc☃')
+    @unittest.skipIf(PY3, 'Python 3 cannot handle bytestrings here.')
```
Comment (ale-rt) [inline src/OFS/tests/testObjectManager.py]: A minor thing I also have been warned about... it is better to use "not PY2" which is more future proof, and we know we have a long future :)

Comment (icemac) [reply src/OFS/tests/testObjectManager.py]: +1 maybe `@unittest.skipUnless(PY2 ...`could be used alternatively to prevent to use the `not`.

Comment (sallner) [reply src/OFS/tests/testObjectManager.py]: I changed this to `skipUnless`. Thanks.

**Comment by icemac** [comment]
@sallner Could you please add a change log entry on master for this PR containing a link to the PR?

**Comment by sallner** [comment]
@icemac Done.

**Pull request merged**